bluetooth/sql/functions/reader_status_history.sql

```diff
@@ -0,0 +1,117 @@
+                     (all_analyses.route_points -> 1) ->> 'id'::text AS end_route_point_id,
+                     (all_analyses.route_points -> 1) ->> 'name'::text AS end_detector
+
+         FROM        status_info
+         INNER JOIN  bluetooth.all_analyses USING (analysis_id))
+
+      -- Create one row for the start and end detector for each analysis_id
+		, detector_list AS (
+         SELECT      route_info.analysis_id,
+                     route_info.start_detector AS detector_name,
+                     route_info.route_status,
+                     route_info.last_reported,
+                     route_info.dt
+         FROM        route_info
+         UNION
+         SELECT      route_info.analysis_id,
+                     route_info.end_detector AS detector_name,
+                     route_info.route_status,
+                     route_info.last_reported,
+			            route_info.dt
+         FROM        route_info)
+
+      -- Select only the detectors that are active
+		, active AS (
+         SELECT      DISTINCT detector_list.detector_name,
+                     max(detector_list.last_reported) AS last_reported,
+                     detector_list.route_status,
+                     detectors_history_corrected.reader_id AS id,
+                     detector_list.dt
+
+         FROM        detector_list
+         LEFT JOIN   bluetooth.detectors_history_corrected ON detector_list.detector_name = detectors_history_corrected.read_name -- Should be changed
+         WHERE       detector_list.route_status = TRUE
+         GROUP BY    detector_list.route_status, detector_list.detector_name, detectors_history_corrected.reader_id, detector_list.dt)
+      
+      -- Combine active and deactive detector 
+      , detector_status as (
+         SELECT      DISTINCT detector_list.detector_name,
+                     max(detector_list.last_reported) AS last_reported,
+                     detector_list.route_status,
+                     detectors_history_corrected.reader_id,
+			            detector_list.dt
+         FROM        detector_list
+         LEFT JOIN   bluetooth.detectors_history_corrected ON detector_list.detector_name = detectors_history_corrected.read_name  -- Should be changed
+         WHERE       detector_list.route_status = FALSE AND 
+                     NOT (detector_list.detector_name IN (SELECT active.detector_name FROM active))
+         GROUP BY    detector_list.route_status, detector_list.detector_name, detectors_history_corrected.reader_id, detector_list.dt
+
+         UNION
+         
+         SELECT      active.detector_name,
+                     active.last_reported,
+                     active.route_status,
+                     active.id,
+                     active.dt
+         FROM        active)
+   
+   INSERT INTO       bluetooth.reader_status_history (reader_id, last_active_date, active, dt)
+
+   SELECT            DISTINCT reader_id, max(last_reported), route_status, dt
```

Comment:
I don't think this select statement is necessary because the previous cte already remove possible dups